python-avd/pyavd/_eos_designs/structured_config/overlay/router_bgp.py

```diff
@@ -24,457 +26,360 @@ class RouterBgpMixin(Protocol):
         ):
-            peer_groups.append({"name": self.inputs.bgp_peer_groups.evpn_overlay_core.name, "activate": False})
+            peer_groups.append_new(name=self.inputs.bgp_peer_groups.evpn_overlay_core.name, activate=False)
 
         if self.shared_utils.overlay_routing_protocol == "ibgp":
             if self.shared_utils.overlay_mpls is True:
-                peer_groups.append({"name": self.inputs.bgp_peer_groups.mpls_overlay_peers.name, "activate": False})
+                peer_groups.append_new(name=self.inputs.bgp_peer_groups.mpls_overlay_peers.name, activate=False)
 
             if self._is_mpls_server is True:
-                peer_groups.append({"name": self.inputs.bgp_peer_groups.rr_overlay_peers.name, "activate": False})
+                peer_groups.append_new(name=self.inputs.bgp_peer_groups.rr_overlay_peers.name, activate=False)
 
         if self.shared_utils.overlay_ipvpn_gateway is True:
-            peer_groups.append({"name": self.inputs.bgp_peer_groups.ipvpn_gateway_peers.name, "activate": False})
-
-        return {"peer_groups": peer_groups}
+            peer_groups.append_new(name=self.inputs.bgp_peer_groups.ipvpn_gateway_peers.name, activate=False)
 
-    def _address_family_evpn(self: AvdStructuredConfigOverlayProtocol) -> dict | None:
-        address_family_evpn = {}
-
-        peer_groups = []
-
-        overlay_peer_group = {}
+    def _set_address_family_evpn(self: AvdStructuredConfigOverlayProtocol) -> None:
+        peer_groups = self.structured_config.router_bgp.address_family_evpn.peer_groups
+        overlay_peer_group = EosCliConfigGen.RouterBgp.AddressFamilyEvpn.PeerGroupsItem()
         if self.shared_utils.is_wan_router:
-            wan_overlay_peer_group = {
-                "name": self.inputs.bgp_peer_groups.wan_overlay_peers.name,
-                "activate": True,
-                "encapsulation": self.inputs.wan_encapsulation,
-            }
+            wan_overlay_peer_group = EosCliConfigGen.RouterBgp.AddressFamilyEvpn.PeerGroupsItem(
+                name=self.inputs.bgp_peer_groups.wan_overlay_peers.name,
+                activate=True,
+                encapsulation=self.inputs.wan_encapsulation,
+            )
             if self.shared_utils.wan_role != "server":
-                wan_overlay_peer_group.update(
-                    {
-                        "route_map_in": "RM-EVPN-SOO-IN",
-                        "route_map_out": "RM-EVPN-SOO-OUT",
-                    },
+                wan_overlay_peer_group._update(
+                    route_map_in="RM-EVPN-SOO-IN",
+                    route_map_out="RM-EVPN-SOO-OUT",
                 )
             peer_groups.append(wan_overlay_peer_group)
 
         if self.shared_utils.overlay_evpn_vxlan is True:
-            overlay_peer_group = {"name": self.inputs.bgp_peer_groups.evpn_overlay_peers.name, "activate": True}
+            overlay_peer_group._update(name=self.inputs.bgp_peer_groups.evpn_overlay_peers.name, activate=True)
```

Comment:
Actually this could also be moved down below the ebgp section since don't need it until the ibgp section.

Reply:
Fixed!